packages/app/src/app/overmind/namespaces/dashboard/actions.ts

```diff
@@ -2158,35 +2143,42 @@ export const removeRepositoryFromTeam = async (
   try {
     await effects.api.removeRepositoryFromTeam(owner, name, teamId);
 
-    if (page === 'recent') {
-      // First, manually remove the data from the state.
-      dashboard.sandboxes.RECENT_BRANCHES =
-        dashboard.sandboxes.RECENT_BRANCHES?.filter(b => {
-          const branchRepo = b.project.repository;
-
-          return branchRepo.owner === owner && branchRepo.name === name;
-        }) ?? [];
+    // Remove all cached data about repository
+    // 1. From repositories list
+    // 2. From repository with branches cache
+    // 3. Branches from recent page
 
-      // Then sync in the background.
-      actions.dashboard.getStartPageSandboxes();
-    } else {
-      // First, manually remove the data from the state.
-      const _repositories = dashboard.repositories ?? {};
-      const filteredRepositoriesByTeam =
-        _repositories[activeTeam]?.filter(
+    const teamRepositories = dashboard.repositoriesByTeamId[activeTeam];
+    if (teamRepositories) {
+      dashboard.repositoriesByTeamId = {
+        ...dashboard.repositoriesByTeamId,
+        [activeTeam]: teamRepositories.filter(
           r => r.repository.owner !== owner || r.repository.name !== name
-        ) ?? [];
-
-      dashboard.repositories = {
-        ..._repositories,
-        [activeTeam]: filteredRepositoriesByTeam,
+        ),
       };
+    }
 
-      // Then sync in the background.
-      actions.dashboard.getRepositoriesByTeam({
-        teamId: activeTeam,
-        bypassLoading: true,
-      });
+    const key = getProjectUniqueKey({ teamId, owner, name });
+    const repositoryWithBranches = dashboard.repositoriesWithBranches[key];
+    if (repositoryWithBranches) {
+      delete dashboard.repositoriesWithBranches[key];
```

Comment:
Does / should this `delete` re-trigger a state update equal to creating a new object?

Reply:
As far as I tested it does work, now that you ask, I don't think this triggers a re-render specifically, but something else does? I will re-test it